### ISSUE TITLE
Feature request: add MAY (Random Maclaurin) and RAY (radial) linear-feature maps for spherical Yat-attention, with bias support

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#36]
### Summary

`nmn.nnx` currently ships a single linear-feature scheme for spherical Yat-attention: the **SLAY anchor method** in `spherical_yat_performer.py` (`create_yat_tp_projection` + `yat_tp_features`) — Bernstein integral → anchor squares for `(q·k)²` → Positive Random Features for `exp(2s·q·k)` → Gauss–Laguerre quadrature → tensor product, `F = R·P·M`.

It approximates **only the bias-free kernel** `K(q,k) = (q·k)² / ((2+ε) − 2 q·k)` (`b = 0`). Deployed Yat-attention learns a **per-head bias `b > 0`** in the numerator `(q·k + b)²`, which SLAY's anchor features cannot represent. I'd like to add two sibling feature maps that do:

- **MAY** — Random Maclaurin features for the on-sphere kernel `κ(s) = (s+b)² / ((2+ε) − 2s)`. A single dot-product map at a freely chosen dimension `M`; supports the bias `b`.
- **RAY** — sketched degree-2 modulation × radial RFF (the off-sphere / sharp-ε route); also bias-aware.

### Why (benchmark)

I compared MAY and RAY against nmn's **actual** SLAY code at matched feature dimension `F`, same linear-attention readout, SLAY given its best `(P,M,R)` split per budget. Metric: per-token cosine similarity / relative-L2 of the linearized attention output vs exact (`d=64`, `N=512`, eps = median sq-distance, 3 seeds).

**The bias `b` is the controlling variable:**

| b | MAY cos / relL2 | RAY cos / relL2 | SLAY cos / relL2 |
|---|---|---|---|
| 0.00 | 0.52 / 1.65 | 0.23 / 3.78 | **0.62 / 0.78** |
| 0.25 | **0.86 / 0.59** | 0.71 / 0.94 | 0.78 / 0.62 |
| 0.50 | **0.96 / 0.28** | 0.88 / 0.49 | 0.89 / 0.45 |
| 1.00 | **0.99 / 0.14** | 0.91 / 0.39 | 0.94 / 0.34 |
| 2.00 | **1.00 / 0.075** | 0.91 / 0.39 | 0.96 / 0.30 |
| 4.00 | **1.00 / 0.049** | 0.91 / 0.39 | 0.96 / 0.28 |

- **The deployment regime is `b > 0`** (trained Yat-attention learns a per-head bias; e.g. our `d22` checkpoint learned `b ≈ 13`). There **MAY is near-exact** — cos `0.99` / relL2 `0.135` at `b = 1`, → `1.00` / `0.05` at `b = 4` — while **SLAY saturates at a wrong-kernel floor** (~`0.94` cos / `0.34` relL2 at `b = 1`, and it does not improve with feature budget) because its `b = 0` anchor features cannot carry the bias. **This is the gap: nmn has no bias-aware linear feature map.**
- `b = 0` is SLAY's design point, and there it is fine (slightly ahead of MAY) — but that is not the regime models train in.

The methods are complementary along the bias axis, and the deployment side (`b > 0`) is the one nmn doesn't yet cover.

### Why `b > 0` is the regime that matters (not just the benchmark)

Yat-attention is an **L1-normalized kernel smoother**, `attn(q) = Σ_j κ(q·k_j) v_j / Σ_j κ(q·k_j)` — same normalization as softmax (sum of weights), different kernel. Two facts make `b > 0` the right target:

- The on-sphere coefficients are `a_n = β_{n-2} + 2b·β_{n-1} + b²·β_n`, so `b > 0` gives `a_n > 0` for **all** `n` (the full positive Maclaurin spectrum, the universal dot-product class that also contains softmax's `exp(q·k)`). At `b = 0`, `a_0 = a_1 = 0`.
- `κ(0) = b²/(2+ε)`. With `b > 0` an orthogonal key gets **positive** weight, so the smoother can represent **diffuse** attention, not only peaked patterns. At `b = 0`, `κ(0) = 0`: orthogonal keys get exactly zero weight and the uniform end is unreachable.

So SLAY's `b = 0` isn't merely behind on the benchmark — it targets the spectrally-deficient kernel that cannot attend diffusely. A bias-aware MAY/RAY feature map covers the regime trained models actually use.

### Proposed API

Slot into the existing performer interface as sibling feature maps:

```python
# nmn/nnx/layers/attention/maclaurin_yat.py
def create_maclaurin_projection(key, head_dim, num_features=512, nmax=40,
                                epsilon=1e-5, bias=1.0, dtype=jnp.float32) -> dict: ...
def maclaurin_features(x, params, normalize=True) -> Array:  # [..., num_features]
    # degrees N ~ a_n/Z (on-sphere coeffs a_n=β_{n-2}+2bβ_{n-1}+b²β_n), Rademacher signs;
    # φ_r(x) = sqrt(Z/M) * prod_l (ω_{r,l}·x_hat)

# nmn/nnx/layers/attention/radial_yat.py  (RAY)
def create_radial_projection(key, head_dim, num_radial=8, sketch_m=128,
                             epsilon=1e-5, bias=1.0, dtype=jnp.float32) -> dict: ...
def radial_features(x, params, normalize=True) -> Array: ...
```

and a selector on the attention layer, e.g. `MultiHeadYatAttention(..., performer_kind="slay" | "maclaurin" | "radial")`, reusing the existing `yat_performer_attention` num/den machinery.

### Notes / caveats to surface in docs

- MAY's Random-Maclaurin features are **sign-indefinite** (no positivity guarantee, unlike SLAY's anchors), so the linear-attention denominator can in principle go non-positive. Empirically it stays well-conditioned at `b > 0`, but the docstring should flag it (and the `b = 0` regime where it degrades).
- MAY's per-feature cost is the mean sampled degree `E[N] = 2/ε + 2/(1+b)`; a degree cap `nmax` bounds cost and regularizes variance at a geometric bias.
- Reference implementations (NumPy) and the full benchmark are available; happy to open a PR porting them to `nnx`.

Methods are from the spherical Yat random-features work (MAY) and the Bernstein–Schur / radial-randomization paper (RAY).
